lib/saluki-app/src/memory.rs

```diff
@@ -68,10 +69,22 @@ impl MemoryBoundsConfiguration {
     ///
     /// If an error occurs during deserialization, an error will be returned.
     pub fn try_from_config(config: &GenericConfiguration) -> Result<Self, GenericError> {
-        let config = config
+        let mut config = config
             .as_typed::<Self>()
             .error_context("Failed to parse memory bounds configuration.")?;
 
+        if config.memory_limit.is_none() {
+            // Try to pull configured memory limit from Cgroup if running in a containerized environment.
+            if let Ok(value) = env::var("DOCKER_DD_AGENT") {
+                if !value.is_empty() {
+                    let cgroup_memory_reader = CgroupMemoryParser;
+                    if let Some(memory) = cgroup_memory_reader.parse() {
+                        config.memory_limit = Some(memory);
```

Comment:
I would probably add an info level log line here saying that we configured the memory limit based on the detected cgroup limit:

```rust
info!("Setting memory limit to {} based on detected cgroups limit.", memory.to_string_as(true));
```